### PR TITLE
Reduce amount of strings we need to pull/push with Transifex

### PR DESCRIPTION
**Pull request by bbondy**
Resolves https://github.com/brave/brave-browser/issues/10186
Sister PR: https://github.com/brave/brave-browser/pull/10560

- Automatically apply branding transformations for both grd and xtb files.
- For strings with incognito -> private, people -> profile, etc, we generate an override file for the grds that have those strings.
- We send up the _override slugs to transifex for translation, I manually applied the old translations to these so we'd have them.
- We pull down the _override xtb files from Transifex.
- We combine those override xtb files into the original ones and remove any matching fingerprint.

For the newly generated xtb and grd updates, see this PR: https://github.com/brave/brave-core/pull/5998

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
  - via npm run pep8 in src/brave
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by bbondy** [comment]
This is updated and ready for another review.
The basic idea is that we:

- Automatically apply branding transformations for both grd and xtb files.
- For strings with incognito -> private, people -> profile, etc, we generate an override file for the grds that have those strings.
- We send up the _override slugs to transifex for translation, I manually applied the old translations to these so we'd have them.
- We pull down the _override xtb files from Transifex.
- We combine those override xtb files into the original ones and remove any matching fingerprint.

The actual XTB, GRD, and GRDP files are in another PR that is against the `l10n-less` branch.

File: script/lib/grd_string_replacements.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    (r'Incognito', r'Private'),
+    (r'inco&gnito', r'&private'),
+    (r'Inco&gnito', r'&Private'),
+    (r'People', r'Profiles'),
+    # 'people' but only in the context of profiles, not humans.
+    (r'(?<!authenticate )people', r'profiles'),
+    (r'(Person)(?!\w)', r'Profile'),
+    (r'(person)(?!\w)', r'profile'),
+    (r'Bookmarks Bar\n', r'Bookmarks\n'),
+    (r'Bookmarks bar\n', r'Bookmarks\n'),
```
Comment (emerick) [inline script/lib/grd_string_replacements.py:39]: Are these two lines in conflict with each other? Or maybe I'm not understanding how this works...

Comment (bbondy) [reply script/lib/grd_string_replacements.py:39]: just case sensitive replacement. Bar vs bar. Could be done with a regex probably too

Comment (emerick) [reply script/lib/grd_string_replacements.py:39]: Ah, missed that!

**Pull request merged**